config/config.go

```diff
@@ -52,30 +52,22 @@ func SetDefaultDirName(binaryName string) {
 // Dir is the configured config home directory.
 // All the cli-related config files live in this directory.
 func Dir() string {
-	var dir string
+	var ev string
 	if runtime.GOOS == "windows" {
-		dir = os.Getenv("APPDATA")
-		if dir != "" {
-			return filepath.Join(dir, DefaultDirName)
-		}
+		ev = os.Getenv("APPDATA")
 	} else {
-		dir := os.Getenv("EXERCISM_CONFIG_HOME")
-		if dir != "" {
-			return dir
-		}
-		dir = os.Getenv("XDG_CONFIG_HOME")
-		if dir == "" {
-			dir = filepath.Join(os.Getenv("HOME"), ".config")
-		}
-		if dir != "" {
-			return filepath.Join(dir, DefaultDirName)
+		if ev = os.Getenv("EXERCISM_CONFIG_HOME"); ev == "" {
+			if ev = os.Getenv("XDG_CONFIG_HOME"); ev == "" {
+				ev = filepath.Join(os.Getenv("HOME"), ".config")
+			}
 		}
 	}
-	// If all else fails, use the current directory.
-	dir, _ = os.Getwd()
-	return dir
+	if ev == ".config" {
```

Comment:
I _think_ this would fix the change for `XDG_CONFIG_HOME`, however it now changes the behavior for windows, right? In order to approve, I think we would need to see a test case that documents the expected config file names for each OS and confirms the previous behavior matches the new behavior. Otherwise, I don't think the refactor is  worth the risk of breaking the configs for some subset of users. \cc @ErikSchierboom

Reply:
> I think we would need to see a test case that documents the expected config file names for each OS and confirms the previous behavior matches the new behavior.

I really wouldn't want a refactoring to change behavior (which technically means it _isn't_ a refactoring).